### ISSUE TITLE
Big files not uploading successfully

### ISSUE THREAD

**Issue by kapilsharma** [cinely/mule-uploader#69]
I hosted Mule Uploader locally.
It is working fine for smaller files but for 9GB file, I'm getting error after it reach 100% and file is not getting saved on S3. Last request in firebug was

http://`mybucket`.s3.amazonaws.com/C8`******`.4Ob4hw`stripped`?uploadId=62`XXXXX`&X-Amz-Date=20160426T115532Z&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Expires=86400&X-Amz-Credential=AKIAJ`<strip>`us-west-2%2Fs3%2Faws4_request&X-Amz-SignedHeaders=host&X-Amz-Signature=e69e`<strip>`

and response XML was

```
<Error>
  <Code>NoSuchUpload</Code>
  <Message>The specified upload does not exist. The upload ID may be invalid, or the upload may have been aborted or completed.</Message>
  <UploadId>62*****</UploadId>
  <RequestId>6C2************</RequestId>
  <HostId>M5W******/45gI5*****************************************</HostId>
</Error>
```

This happens only with big 9GB file. For smaller files (below 1 GB), everything is working fine.

Please let me know if more details needed on it.

**Comment by volkeransmann** [comment]
Having the exact same issue. But happens not always. Just tried with 30GB - working fine. 60GB will not finish. (onComplete does not fire). Upon trying to resume i get the same error xml.
The file is not showing in S3. Any Ideas?

**Comment by volkeransmann** [comment]
I think I have a clue here: 
The ListPartsResult that is fetched in order to resume the upload has a MaxParts limit of 1.000.
So if you have more than 1.000 parts then you are supposed to do a "subsequent list parts request".
I thin this is not happending at the moment. Maybe this is the Problem?

See here: http://docs.aws.amazon.com/AmazonS3/latest/API/mpUploadListParts.html

**Comment by abuisine** [comment]
solved by 24863a2f86429353ef729640ab25b41a73fff543
check version 1.1.3 ;)
